utils/types.go

```diff
@@ -18,3 +20,24 @@ type CredStruct struct {
 	Password string `json:"password"`
 	Customer string `json:"customer"`
 }
+
+type Job struct {
+	ctx        context.Context
+	sessionObj SessionObj
+}
+
+func (j *Job) GetContext() context.Context {
+	return j.ctx
+}
+
+func (j *Job) GetObj() SessionObj {
```

Comment:
nitpick: getters in Go [should not](https://go.dev/doc/effective_go#Getters) be named `Get`

>  If you have a field called owner (lower case, unexported), the getter method should be called Owner (upper case, exported), not GetOwner.

```suggestion
func (j *Job) Context() context.Context {
	return j.ctx
}

func (j *Job) Obj() SessionObj {
```